pkg/ccl/logictestccl/testdata/logic_test/super_regions

```diff
@@ -0,0 +1,654 @@
+                                                                range_max_bytes = 536870912,
+                                                                gc.ttlseconds = 90000,
+                                                                num_replicas = 4,
+                                                                num_voters = 3,
+                                                                constraints = '{+region=ap-southeast-2: 1, +region=us-east-1: 1}',
+                                                                voter_constraints = '[+region=ap-southeast-2]',
+                                                                lease_preferences = '[[+region=ap-southeast-2]]'
+
+# Check that indexes also have the updated zone configs.
+query TT
+SHOW ZONE CONFIGURATION FOR PARTITION "ap-southeast-2" OF INDEX mr2.t_rbr_with_idx2@idx
+----
+PARTITION "ap-southeast-2" OF INDEX mr2.public.t_rbr_with_idx2@idx  ALTER PARTITION "ap-southeast-2" OF INDEX mr2.public.t_rbr_with_idx2@idx CONFIGURE ZONE USING
+                                                                    range_min_bytes = 134217728,
+                                                                    range_max_bytes = 536870912,
+                                                                    gc.ttlseconds = 90000,
+                                                                    num_replicas = 4,
+                                                                    num_voters = 3,
+                                                                    constraints = '{+region=ap-southeast-2: 1, +region=us-east-1: 1}',
+                                                                    voter_constraints = '[+region=ap-southeast-2]',
+                                                                    lease_preferences = '[[+region=ap-southeast-2]]'
+
+query TT
+SHOW ZONE CONFIGURATION FOR PARTITION "us-east-1" OF INDEX mr2.t_rbr_with_idx2@idx
+----
+PARTITION "us-east-1" OF INDEX mr2.public.t_rbr_with_idx2@idx  ALTER PARTITION "us-east-1" OF INDEX mr2.public.t_rbr_with_idx2@idx CONFIGURE ZONE USING
+                                                               range_min_bytes = 134217728,
+                                                               range_max_bytes = 536870912,
+                                                               gc.ttlseconds = 90000,
+                                                               num_replicas = 4,
+                                                               num_voters = 3,
+                                                               constraints = '{+region=ap-southeast-2: 1, +region=us-east-1: 1}',
+                                                               voter_constraints = '[+region=us-east-1]',
+                                                               lease_preferences = '[[+region=us-east-1]]'
+
+# "ca-central-1" is not part of any super region so replicas should be placed in
```

Comment:
This is another UX point I wanted to bring up. @awoods187 cc @ajstorm 

In this case, I chose to implement it such that if the region is not part of a super region, then we default back to the old behaviour by placing it in every region.

Adam suggested that this might be unintuitive and instead we may want to create an "implicit" super region which includes all the regions that are _not_ in any super region and in this case, we place the region in the implicit super region. This may have clarity issues as well.